services/hanke-service/src/main/kotlin/fi/hel/haitaton/hanke/application/ApplicationService.kt

```diff
@@ -216,6 +216,10 @@ open class ApplicationService(
         return applicationRepository.save(application).toApplication()
     }
 
+    open fun sendAttachment(alluId: Int, attachment: Attachment) {
```

Comment:
Couldn't ApplicationAttachmentService call the Allu client itself? Routing attachment updates through ApplicationService feels like mixing their responsibilities.

Maybe instead, sendAllAttachments could be moved to ApplicationAttachmentService, so that all Allu operations on attachments are done there.

Reply:
I think using applicationservice in the middle is ok because:
1. These are attachments of an application.
2. We already have applicationservice injected. Using cablereportservice (how about kaivuuilmoitus etc.) here will bring another dependency to attachments.

So I was thinking that talking to application service is the simplest solution.

Reply:
Sorry, I was not clear enough previously. I meant that if we are going to use isStillPending from applicationservice (suggested in another comment), then we already have applicationservice at use. To me it is confusing to use cablereportservice(send) and applicationservice (isstillpending) to communicate with allu.

**But** if we are going to refactor cablereportservice, we can proceed as you suggested.

Reply:
I did some fixes. I chose to keep sendAllAttachments in applicationservice as a private functions because the use case there is to send attachments along with the application for the first time.

If this method was moved to applicationattachmentservice, then we would have a circular dependency of applicationservice calling attachmentservice and vice versa.

Reply:
```
open class ApplicationService(
    private val applicationRepository: ApplicationRepository,
    private val alluStatusRepository: AlluStatusRepository,
    private val cableReportService: CableReportService,
    private val disclosureLogService: DisclosureLogService,
    private val applicationLoggingService: ApplicationLoggingService,
    private val hankeKayttajaService: HankeKayttajaService,
    private val emailSenderService: EmailSenderService,
    private val geometriatDao: GeometriatDao,
    private val permissionService: PermissionService,
    private val hankeRepository: HankeRepository,
) {
```

This also is starting to look like we should be removing dependencies and not add them.

Reply:
Attachment handling moved to attachment service. I believe this is more like what you suggested?